mgmt/rest/v2/plugin.go

```diff
@@ -0,0 +1,412 @@
+			return
+		}
+		rp.SetSignature(signature)
+		restLogger.Info("Loading plugin: ", rp.Path())
+		pl, err := s.metricManager.Load(rp)
+		if err != nil {
+			var ec int
+			restLogger.Error(err)
+			restLogger.Debugf("Removing file (%s)", rp.Path())
+			err2 := os.RemoveAll(filepath.Dir(rp.Path()))
+			if err2 != nil {
+				restLogger.Error(err2)
+			}
+			rb := response.FromError(err)
+			switch rb.ErrorMessage {
+			case PluginAlreadyLoaded:
+				ec = 409
+			default:
+				ec = 500
+			}
+			response.Write(ec, rb, w)
+			return
+		}
+		response.Write(201, catalogedPluginBody(r.Host, pl), w)
+	}
+}
+
+func writeFile(filename string, b []byte) (string, error) {
+	// Create temporary directory
+	dir, err := ioutil.TempDir("", "")
+	if err != nil {
+		return "", err
+	}
+	f, err := os.Create(path.Join(dir, filename))
+	if err != nil {
+		return "", err
+	}
+	// Close before load
+	defer f.Close()
+
+	n, err := f.Write(b)
+	log.Debugf("wrote %v to %v", n, f.Name())
+	if err != nil {
+		return "", err
+	}
+	if runtime.GOOS != "windows" {
+		err = f.Chmod(0700)
+		if err != nil {
+			return "", err
+		}
+	}
+	return f.Name(), nil
+}
+
+func (s *V2) unloadPlugin(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+	plName := p.ByName("name")
+	plType := p.ByName("type")
+	plVersion, err := strconv.ParseInt(p.ByName("version"), 10, 0)
+	f := map[string]interface{}{
+		"plugin-name":    plName,
+		"plugin-version": plVersion,
+		"plugin-type":    plType,
+	}
+
+	if err != nil || plName == "" || plType == "" {
+		se := serror.New(errors.New("missing or invalid parameter(s)"))
+		se.SetFields(f)
+		response.Write(400, response.FromSnapError(se), w)
+		return
+	}
+
+	_, se := s.metricManager.Unload(&plugin{
+		name:       plName,
+		version:    int(plVersion),
+		pluginType: plType,
+	})
+
+	// 404 - plugin not found
+	// 409 - plugin state is not plugin loaded
+	// 500 - removing plugin from /tmp failed
+	if se != nil {
+		se.SetFields(f)
+		statusCode := 500
+		switch se.Error() {
+		case control.ErrPluginNotFound.Error():
+			statusCode = 404
+		case control.ErrPluginNotInLoadedState.Error():
+			statusCode = 409
+		}
+		response.Write(statusCode, response.FromSnapError(se), w)
+		return
+	}
+	response.Write(204, nil, w)
+}
+
+func (s *V2) getPlugins(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+
+	// filter by plugin name or plugin type
+	plName := params.ByName("name")
+	plType := params.ByName("type")
+	nbFilter := Btoi(plName != "") + Btoi(plType != "")
+
+	if _, detail := r.URL.Query()["running"]; detail {
+		// get running plugins
+		plugins := runningPluginsBody(r.Host, s.metricManager.AvailablePlugins())
+		filteredPlugins := []response.RunningPlugin{}
+		if nbFilter > 0 {
+			for _, p := range plugins {
+				if nbFilter == 1 && (p.Name == plName || p.Type == plType) || nbFilter == 2 && (p.Name == plName && p.Type == plType) {
+					filteredPlugins = append(filteredPlugins, p)
+				}
+			}
+		} else {
```

Comment:
If you move that to the top it'll return every plugins + the append of the filtered plugins.